en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,127 @@
+Block parameter
+===============
+
+Context
+-------
+
+For their display to be customizable, some blocks need to receive parameters when they render.
+Those parameters could be of multiple types, for instance :
+
+- url parameters
+- server parameters
+- header parameters
+- ...
+
+An simple solution to give those parameters to the block would have been to give all the request
+parameters when the block is called.
+
+This solution is not optimal for a main reason: when loading blocks via `ESI`_, each
+block will be cached for each url variant.
+
+If the block 1, which does not require any parameters, will be cached for each url variant :
```

Comment:
You still have a problem with that "If" not followed by a "Then". You could substitute If by "In this example, the block 1 ..."